x-pack/plugins/siem/public/lib/track_usage/index.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { trackUiMetric } from '../../../../../../src/legacy/core_plugins/ui_metric/public';
+
+const APP_NAME = 'siem';
+
+export function trackUiAction(metricType: string) {
+  trackUiMetric(APP_NAME, metricType);
+}
```

Comment:
`APP_NAME` can be pulled in above this like so:

```ts
import { APP_NAME } from '../../..';
```

Also, we typically try not to write named functions, but rather the newer style functions so there aren't problems with `this`

```ts
export const trackUiAction = (metricType: string) => trackUiMetric(APP_NAME, metricType);
```

Altogether it would look like this:

```ts
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { trackUiMetric } from '../../../../../../src/legacy/core_plugins/ui_metric/public';
import { APP_NAME } from '../../..';

export const trackUiAction = (metricType: string) => trackUiMetric(APP_NAME, metricType);
```

Reply:
That looks a bit magic to me, but ok :)

Reply:
The `import { APP_NAME } from '../../..';` results in `SIEM` which I find a bit weird then in JSON docs. I think I'd rather stick with the lower case version.

Reply:
I took the other suggestion.

Reply:
FWIW
The `../../../` is standard inclusion of `index.ts` files. This one points here:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/siem/index.ts#L14

You can optionally use the APP_ID from here:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/siem/index.ts#L13

You would just have to export it:

```ts
export const APP_ID = 'siem';
```

And then include it like so:

```ts
import { APP_ID } from '../../..';
```

Leaving this as optional.

Reply:
I like this, thanks!